saleor/payment/gateways/stripe/webhooks.py

```diff
@@ -158,6 +130,15 @@ def _finalize_checkout(
     )
 
 
+def _get_checkout(payment_id: int) -> Optional[Checkout]:
+    return (
+        Checkout.objects.prefetch_related("payments")
+        .select_for_update(of=("self",))
+        .filter(payments__id=payment_id, payments__is_active=True)
```

Comment:
For processing a notification we should also take into account the payments which are not active